### ISSUE TITLE
downloader issue for git source (without git in the url)

### ISSUE THREAD

**Issue by satmandu** [chromebrew/chromebrew#7278]
```
Package to be built is: crosvm.rb
# Adapted from Arch Linux crosvm-git PKGBUILD at:
# https://aur.archlinux.org/cgit/aur.git/plain/PKGBUILD?h=crosvm-git

require 'package'

class Crosvm < Package
  description 'The Chrome OS Virtual Machine Monitor'
  homepage 'https://chromium.googlesource.com/crosvm/crosvm'
  version '7b5f6b1'
  compatibility 'all'
  license 'custom:chromiumos'
  source_url 'https://chromium.googlesource.com/crosvm/crosvm'
  git_hashtag '7b5f6b198fa2e3acc797175670fad8686db3e72e'

  depends_on 'libcap'
  depends_on 'dtc'
  depends_on 'rust' => :build
  depends_on 'wayland_protocols' => :build

  def self.build
    system 'cargo build --release'
  end

  def self.install
    system "install -Dm755 target/release/crosvm #{CREW_DEST_PREFIX}/bin/crosvm"
    system "install -d #{CREW_DEST_PREFIX}/share/policy/crosvm/"
    system "cp -r seccomp/x86_64/* #{CREW_DEST_PREFIX}/share/policy/crosvm/"
    system "install -m644 -D LICENSE #{CREW_DEST_PREFIX}/share/licenses/crosvm-git/LICENSE"
    system "install -m644 -D README.md #{CREW_DEST_PREFIX}/share/licenses/crosvm-git/README.md"
  end
end
/usr/local/lib/crew/lib/downloader.rb:64:in `downloader': undefined method `green' for nil:NilClass (NoMethodError)

          Expected checksum (SHA256): #{sha256sum.green}
                                                 ^^^^^^
        from /usr/local/bin/crew:744:in `block in download'
        from /usr/local/bin/crew:688:in `chdir'
        from /usr/local/bin/crew:688:in `download'
        from /usr/local/bin/crew:1484:in `build_package'
        from /usr/local/bin/crew:1465:in `resolve_dependencies_and_build'
        from /usr/local/bin/crew:1729:in `block in build_command'
        from /usr/local/bin/crew:1725:in `each'
        from /usr/local/bin/crew:1725:in `build_command'
        from /usr/local/bin/crew:1908:in `<main>'
```

@supechicken ideas?

**Comment by satmandu** [comment]
Changing the `source_url` to `https://chromium.googlesource.com/crosvm/crosvm.git` fixes the issue, but maybe that should be automatic if we have a `git_hashtag` specified?

**Comment by supechicken** [comment]
The `git` URL detect logic is dumb (it requires the URL matches `^git://` or `.git$`)

> Changing the `source_url` to `https://chromium.googlesource.com/crosvm/crosvm.git` fixes the issue, but maybe that should be automatic if we have a `git_hashtag` specified?

Good idea!

**Comment by Bdieir** [comment]
> ```
> Package to be built is: crosvm.rb
> # Adapted from Arch Linux crosvm-git PKGBUILD at:
> # https://aur.archlinux.org/cgit/aur.git/plain/PKGBUILD?h=crosvm-git
> 
> require 'package'
> 
> class Crosvm < Package
>   description 'The Chrome OS Virtual Machine Monitor'
>   homepage 'https://chromium.googlesource.com/crosvm/crosvm'
>   version '7b5f6b1'
>   compatibility 'all'
>   license 'custom:chromiumos'
>   source_url 'https://chromium.googlesource.com/crosvm/crosvm'
>   git_hashtag '7b5f6b198fa2e3acc797175670fad8686db3e72e'
> 
>   depends_on 'libcap'
>   depends_on 'dtc'
>   depends_on 'rust' => :build
>   depends_on 'wayland_protocols' => :build
> 
>   def self.build
>     system 'cargo build --release'
>   end
> 
>   def self.install
>     system "install -Dm755 target/release/crosvm #{CREW_DEST_PREFIX}/bin/crosvm"
>     system "install -d #{CREW_DEST_PREFIX}/share/policy/crosvm/"
>     system "cp -r seccomp/x86_64/* #{CREW_DEST_PREFIX}/share/policy/crosvm/"
>     system "install -m644 -D LICENSE #{CREW_DEST_PREFIX}/share/licenses/crosvm-git/LICENSE"
>     system "install -m644 -D README.md #{CREW_DEST_PREFIX}/share/licenses/crosvm-git/README.md"
>   end
> end
> /usr/local/lib/crew/lib/downloader.rb:64:in `downloader': undefined method `green' for nil:NilClass (NoMethodError)
> 
>           Expected checksum (SHA256): #{sha256sum.green}
>                                                  ^^^^^^
>         from /usr/local/bin/crew:744:in `block in download'
>         from /usr/local/bin/crew:688:in `chdir'
>         from /usr/local/bin/crew:688:in `download'
>         from /usr/local/bin/crew:1484:in `build_package'
>         from /usr/local/bin/crew:1465:in `resolve_dependencies_and_build'
>         from /usr/local/bin/crew:1729:in `block in build_command'
>         from /usr/local/bin/crew:1725:in `each'
>         from /usr/local/bin/crew:1725:in `build_command'
>         from /usr/local/bin/crew:1908:in `<main>'
> ```
> 
> 
>      
>   
> 
> @supechicken ideas?

**Comment by Zopolis4** [comment]
This is solved by #5933, right?

**Comment by Zopolis4** [comment]
Actually, it seems `crew download -s libdbusmenu_gtk3` fails:
![image](https://github.com/chromebrew/chromebrew/assets/64618338/979f826e-4ce5-4ef7-953f-f25817258e7f)
